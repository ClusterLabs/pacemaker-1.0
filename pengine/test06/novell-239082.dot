 digraph "g" {
"all_stopped" [ style=bold color="green" fontcolor="orange"  ]
"do_shutdown xen-1" [ style=bold color="green" fontcolor="black"  ]
"drbd0:0_post_notify_demote_0 xen-2" -> "ms-drbd0_confirmed-post_notify_demote_0" [ style = bold]
"drbd0:0_post_notify_demote_0 xen-2" [ style=bold color="green" fontcolor="black"  ]
"drbd0:0_post_notify_promote_0 xen-2" -> "ms-drbd0_confirmed-post_notify_promote_0" [ style = bold]
"drbd0:0_post_notify_promote_0 xen-2" [ style=bold color="green" fontcolor="black"  ]
"drbd0:0_post_notify_stop_0 xen-2" -> "ms-drbd0_confirmed-post_notify_stop_0" [ style = bold]
"drbd0:0_post_notify_stop_0 xen-2" [ style=bold color="green" fontcolor="black"  ]
"drbd0:0_pre_notify_demote_0 xen-2" -> "ms-drbd0_confirmed-pre_notify_demote_0" [ style = bold]
"drbd0:0_pre_notify_demote_0 xen-2" [ style=bold color="green" fontcolor="black"  ]
"drbd0:0_pre_notify_promote_0 xen-2" -> "ms-drbd0_confirmed-pre_notify_promote_0" [ style = bold]
"drbd0:0_pre_notify_promote_0 xen-2" [ style=bold color="green" fontcolor="black"  ]
"drbd0:0_pre_notify_stop_0 xen-2" -> "ms-drbd0_confirmed-pre_notify_stop_0" [ style = bold]
"drbd0:0_pre_notify_stop_0 xen-2" [ style=bold color="green" fontcolor="black"  ]
"drbd0:0_promote_0 xen-2" -> "ms-drbd0_promoted_0" [ style = bold]
"drbd0:0_promote_0 xen-2" [ style=bold color="green" fontcolor="black"  ]
"drbd0:1_demote_0 xen-1" -> "drbd0:1_stop_0 xen-1" [ style = bold]
"drbd0:1_demote_0 xen-1" -> "ms-drbd0_demoted_0" [ style = bold]
"drbd0:1_demote_0 xen-1" [ style=bold color="green" fontcolor="black"  ]
"drbd0:1_post_notify_demote_0 xen-1" -> "ms-drbd0_confirmed-post_notify_demote_0" [ style = bold]
"drbd0:1_post_notify_demote_0 xen-1" [ style=bold color="green" fontcolor="black"  ]
"drbd0:1_pre_notify_demote_0 xen-1" -> "ms-drbd0_confirmed-pre_notify_demote_0" [ style = bold]
"drbd0:1_pre_notify_demote_0 xen-1" [ style=bold color="green" fontcolor="black"  ]
"drbd0:1_pre_notify_stop_0 xen-1" -> "ms-drbd0_confirmed-pre_notify_stop_0" [ style = bold]
"drbd0:1_pre_notify_stop_0 xen-1" [ style=bold color="green" fontcolor="black"  ]
"drbd0:1_stop_0 xen-1" -> "all_stopped" [ style = bold]
"drbd0:1_stop_0 xen-1" -> "do_shutdown xen-1" [ style = bold]
"drbd0:1_stop_0 xen-1" -> "ms-drbd0_stopped_0" [ style = bold]
"drbd0:1_stop_0 xen-1" [ style=bold color="green" fontcolor="black"  ]
"fs_1_start_0 xen-2" [ style=bold color="green" fontcolor="black"  ]
"fs_1_stop_0 xen-1" -> "all_stopped" [ style = bold]
"fs_1_stop_0 xen-1" -> "do_shutdown xen-1" [ style = bold]
"fs_1_stop_0 xen-1" -> "fs_1_start_0 xen-2" [ style = bold]
"fs_1_stop_0 xen-1" -> "ms-drbd0_demote_0" [ style = bold]
"fs_1_stop_0 xen-1" [ style=bold color="green" fontcolor="black"  ]
"ms-drbd0_confirmed-post_notify_demote_0" -> "ms-drbd0_pre_notify_stop_0" [ style = bold]
"ms-drbd0_confirmed-post_notify_demote_0" -> "ms-drbd0_stop_0" [ style = bold]
"ms-drbd0_confirmed-post_notify_demote_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_confirmed-post_notify_promote_0" -> "fs_1_start_0 xen-2" [ style = bold]
"ms-drbd0_confirmed-post_notify_promote_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_confirmed-post_notify_stop_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_confirmed-pre_notify_demote_0" -> "ms-drbd0_demote_0" [ style = bold]
"ms-drbd0_confirmed-pre_notify_demote_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_confirmed-pre_notify_promote_0" -> "ms-drbd0_promote_0" [ style = bold]
"ms-drbd0_confirmed-pre_notify_promote_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_confirmed-pre_notify_stop_0" -> "ms-drbd0_stop_0" [ style = bold]
"ms-drbd0_confirmed-pre_notify_stop_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_demote_0" -> "drbd0:1_demote_0 xen-1" [ style = bold]
"ms-drbd0_demote_0" -> "ms-drbd0_demoted_0" [ style = bold]
"ms-drbd0_demote_0" -> "ms-drbd0_stop_0" [ style = bold]
"ms-drbd0_demote_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_demoted_0" -> "ms-drbd0_post_notify_demote_0" [ style = bold]
"ms-drbd0_demoted_0" -> "ms-drbd0_promote_0" [ style = bold]
"ms-drbd0_demoted_0" -> "ms-drbd0_stop_0" [ style = bold]
"ms-drbd0_demoted_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_post_notify_demote_0" -> "drbd0:0_post_notify_demote_0 xen-2" [ style = bold]
"ms-drbd0_post_notify_demote_0" -> "drbd0:1_post_notify_demote_0 xen-1" [ style = bold]
"ms-drbd0_post_notify_demote_0" -> "ms-drbd0_confirmed-post_notify_demote_0" [ style = bold]
"ms-drbd0_post_notify_demote_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_post_notify_promote_0" -> "drbd0:0_post_notify_promote_0 xen-2" [ style = bold]
"ms-drbd0_post_notify_promote_0" -> "ms-drbd0_confirmed-post_notify_promote_0" [ style = bold]
"ms-drbd0_post_notify_promote_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_post_notify_stop_0" -> "drbd0:0_post_notify_stop_0 xen-2" [ style = bold]
"ms-drbd0_post_notify_stop_0" -> "ms-drbd0_confirmed-post_notify_stop_0" [ style = bold]
"ms-drbd0_post_notify_stop_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_pre_notify_demote_0" -> "drbd0:0_pre_notify_demote_0 xen-2" [ style = bold]
"ms-drbd0_pre_notify_demote_0" -> "drbd0:1_pre_notify_demote_0 xen-1" [ style = bold]
"ms-drbd0_pre_notify_demote_0" -> "ms-drbd0_confirmed-pre_notify_demote_0" [ style = bold]
"ms-drbd0_pre_notify_demote_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_pre_notify_promote_0" -> "drbd0:0_pre_notify_promote_0 xen-2" [ style = bold]
"ms-drbd0_pre_notify_promote_0" -> "ms-drbd0_confirmed-pre_notify_promote_0" [ style = bold]
"ms-drbd0_pre_notify_promote_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_pre_notify_stop_0" -> "drbd0:0_pre_notify_stop_0 xen-2" [ style = bold]
"ms-drbd0_pre_notify_stop_0" -> "drbd0:1_pre_notify_stop_0 xen-1" [ style = bold]
"ms-drbd0_pre_notify_stop_0" -> "ms-drbd0_confirmed-pre_notify_stop_0" [ style = bold]
"ms-drbd0_pre_notify_stop_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_promote_0" -> "drbd0:0_promote_0 xen-2" [ style = bold]
"ms-drbd0_promote_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_promoted_0" -> "fs_1_start_0 xen-2" [ style = bold]
"ms-drbd0_promoted_0" -> "ms-drbd0_post_notify_promote_0" [ style = bold]
"ms-drbd0_promoted_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_stop_0" -> "drbd0:1_stop_0 xen-1" [ style = bold]
"ms-drbd0_stop_0" -> "ms-drbd0_stopped_0" [ style = bold]
"ms-drbd0_stop_0" [ style=bold color="green" fontcolor="orange"  ]
"ms-drbd0_stopped_0" -> "ms-drbd0_post_notify_stop_0" [ style = bold]
"ms-drbd0_stopped_0" -> "ms-drbd0_promote_0" [ style = bold]
"ms-drbd0_stopped_0" [ style=bold color="green" fontcolor="orange"  ]
}
