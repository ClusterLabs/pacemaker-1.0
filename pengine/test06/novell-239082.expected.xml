<?xml version="1.0" encoding="UTF-8"?>
<cib admin_epoch="0" epoch="16" num_updates="440" dc-uuid="b292356d-d8af-4360-bcd3-423b3ec7f1be" have-quorum="true" remote-tls-port="0" validate-with="pacemaker-1.0">
   <configuration>
     <crm_config>
       <cluster_property_set id="cib-bootstrap-options"><nvpair id="nvpair" name="default-action-timeout" value="120s"/><nvpair id="nvpair" name="cluster-delay" value="60s"/><nvpair id="nvpair" name="symmetric-cluster" value="true"/><nvpair id="nvpair" name="stop-orphan-resources" value="true"/><nvpair id="nvpair" name="stop-orphan-actions" value="true"/><nvpair id="nvpair" name="remove-after-stop" value="false"/><nvpair id="nvpair" name="is-managed-default" value="true"/><nvpair id="nvpair" name="no-quorum-policy" value="ignore"/><nvpair id="nvpair" name="stonith-action" value="reboot"/><nvpair id="nvpair" name="stonith-enabled" value="0"/><nvpair id="nvpair" name="pe-error-series-max" value="100"/><nvpair id="nvpair" name="pe-warn-series-max" value="100"/><nvpair id="nvpair" name="pe-input-series-max" value="100"/><nvpair id="nvpair" name="default-resource-stickiness" value="100"/><nvpair id="nvpair" name="default-migration-threshold" value="0"/><nvpair id="nvpair" name="shutdown-escalation" value="5min"/><nvpair id="nvpair" name="startup-fencing" value="false"/><nvpair id="nvpair" name="last-lrm-refresh" value="1169747681"/></cluster_property_set>
     </crm_config>
     <nodes>
       <node id="b292356d-d8af-4360-bcd3-423b3ec7f1be" uname="xen-2" type="normal"/>
       <node id="00aaa02f-110f-44e6-9a9d-af93827bfdb9" uname="xen-1" type="normal"/>
     </nodes>
     <resources>
       <primitive id="fs_1" class="ocf" provider="heartbeat" type="Filesystem"><meta_attributes id="primitive-fs_1.meta"><nvpair id="nvpair" name="target-role" value="#default"/></meta_attributes>
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="fstype" value="reiserfs"/><nvpair value="/mnt/share1" id="nvpair" name="directory"/><nvpair id="nvpair" name="device" value="/dev/drbd0"/><nvpair name="target_role" id="nvpair" value="#default"/></instance_attributes>
       </primitive>
       <master id="ms-drbd0"><meta_attributes id="master_slave-ms-drbd0.meta"/><meta_attributes id="meta_attributes"><nvpair id="nvpair" name="clone-max" value="2"/><nvpair id="nvpair" name="clone-node-max" value="1"/><nvpair id="nvpair" name="master-max" value="1"/><nvpair id="nvpair" name="master-node-max" value="1"/><nvpair id="nvpair" name="notify" value="yes"/><nvpair id="nvpair" name="globally-unique" value="false"/><nvpair id="nvpair" name="target-role" value="#default"/></meta_attributes><primitive id="drbd0" class="ocf" provider="heartbeat" type="drbd"><meta_attributes id="primitive-drbd0.meta"/>
           <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="drbd_resource" value="r0"/><nvpair id="nvpair" name="clone_overrides_hostname" value="no"/></instance_attributes>
         </primitive></master>
     </resources>
     <constraints>
       <rsc_order id="drbd_before_fs" first="ms-drbd0" then="fs_1" then-action="start" first-action="promote" score="0"/>
       <rsc_colocation id="fs_on_drbd0" rsc="fs_1" with-rsc="ms-drbd0" with-rsc-role="Master" score="INFINITY"/>
     </constraints>
   </configuration>
   <status>
     <node_state id="b292356d-d8af-4360-bcd3-423b3ec7f1be" uname="xen-2" crmd="online" shutdown="0" in_ccm="true" ha="active" join="member" expected="member">
       <lrm id="b292356d-d8af-4360-bcd3-423b3ec7f1be">
         <lrm_resources id="lrm_resources">
           <lrm_resource id="fs_1" type="Filesystem" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="fs_1_monitor_0" operation="monitor" transition-key="3:0:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:7;3:0:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="2" crm_feature_set="1.0.8" rc-code="7" op-status="0" interval="0" op-digest="e38fc095cd72534897b21e24a8484122"/>
           </lrm_resource>
           <lrm_resource id="drbd0:0" type="drbd" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="drbd0-0_monitor_0" operation="monitor" transition-key="4:0:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:7;4:0:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="3" crm_feature_set="1.0.8" rc-code="7" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
             <lrm_rsc_op id="drbd0-0_start_0" operation="start" transition-key="4:2:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:0;4:2:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="5" crm_feature_set="1.0.8" rc-code="0" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
             <lrm_rsc_op id="drbd0-0_post_notify_start_0" operation="notify" transition-key="30:2:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:0;30:2:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="6" crm_feature_set="1.0.8" rc-code="0" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
             <lrm_rsc_op id="drbd0-0_pre_notify_promote_0" operation="notify" transition-key="46:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:0;46:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="7" crm_feature_set="1.0.8" rc-code="0" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
             <lrm_rsc_op id="drbd0-0_post_notify_promote_0" operation="notify" transition-key="47:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:0;47:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="8" crm_feature_set="1.0.8" rc-code="0" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
           </lrm_resource>
           <lrm_resource id="drbd0:1" type="drbd" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="drbd0-1_monitor_0" operation="monitor" transition-key="3:1:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:7;3:1:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="4" crm_feature_set="1.0.8" rc-code="7" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
           </lrm_resource>
         </lrm_resources>
       </lrm>
       <transient_attributes id="b292356d-d8af-4360-bcd3-423b3ec7f1be">
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="probe_complete" value="true"/></instance_attributes>
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="master-drbd0:0" value="10"/></instance_attributes>
       </transient_attributes>
     </node_state>
     <node_state id="00aaa02f-110f-44e6-9a9d-af93827bfdb9" uname="xen-1" crmd="online" shutdown="1169750793" in_ccm="true" ha="active" join="member" expected="down">
       <lrm id="lrm.auto-2">
         <lrm_resources id="lrm_resources">
           <lrm_resource id="fs_1" type="Filesystem" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="fs_1_monitor_0" operation="monitor" transition-key="6:0:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:7;6:0:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="2" crm_feature_set="1.0.8" rc-code="7" op-status="0" interval="0" op-digest="e38fc095cd72534897b21e24a8484122"/>
             <lrm_rsc_op id="fs_1_start_0" operation="start" transition-key="4:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:0;4:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="10" crm_feature_set="1.0.8" rc-code="0" op-status="0" interval="0" op-digest="e38fc095cd72534897b21e24a8484122"/>
           </lrm_resource>
           <lrm_resource id="drbd0:0" type="drbd" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="drbd0-0_monitor_0" operation="monitor" transition-key="7:0:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:7;7:0:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="3" crm_feature_set="1.0.8" rc-code="7" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
           </lrm_resource>
           <lrm_resource id="drbd0:1" type="drbd" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="drbd0-1_monitor_0" operation="monitor" transition-key="5:1:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:7;5:1:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="4" crm_feature_set="1.0.8" rc-code="7" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
             <lrm_rsc_op id="drbd0-1_start_0" operation="start" transition-key="5:2:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:0;5:2:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="5" crm_feature_set="1.0.8" rc-code="0" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
             <lrm_rsc_op id="drbd0-1_post_notify_start_0" operation="notify" transition-key="33:2:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:0;33:2:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="6" crm_feature_set="1.0.8" rc-code="0" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
             <lrm_rsc_op id="drbd0-1_pre_notify_promote_0" operation="notify" transition-key="38:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:0;38:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="7" crm_feature_set="1.0.8" rc-code="0" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
             <lrm_rsc_op id="drbd0-1_promote_0" operation="promote" transition-key="7:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:0;7:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="8" crm_feature_set="1.0.8" rc-code="0" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
             <lrm_rsc_op id="drbd0-1_post_notify_promote_0" operation="notify" transition-key="39:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" transition-magic="0:0;39:3:c3ac9c9a-8ef7-46c7-9075-996d77e06870" call-id="9" crm_feature_set="1.0.8" rc-code="0" op-status="0" interval="0" op-digest="7ed7a3f295075a81edf4174cf92234c6"/>
           </lrm_resource>
         </lrm_resources>
       </lrm>
       <transient_attributes id="transient_attributes.auto-2">
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="probe_complete" value="true"/></instance_attributes>
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="master-drbd0:1" value="75"/></instance_attributes>
       </transient_attributes>
     </node_state>
   </status>
 </cib>
